gazebo_plugins/src/gazebo_ros_template.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/rclcpp.hpp>
+#include <gazebo_ros/node.hpp>
+
+#include <gazebo/physics/Model.hh>
+
+#include <gazebo_plugins/gazebo_ros_template.hpp>
+
+namespace gazebo
+{
+/// Class to hold private data members (PIMPL pattern)
+class GazeboRosTemplatePrivate
+{
+public:
+  /// Connection to world update event. Callback is called while this is alive.
+  event::ConnectionPtr update_connection_;
+
+  /// Node for ROS communication.
+  gazebo_ros::Node::SharedPtr ros_node_;
+};
+
+GazeboRosTemplate::GazeboRosTemplate()
+: d_ptr_(new GazeboRosTemplatePrivate)
```

Comment:
```make_unique``` is slightly cleaner I think

Reply:
making unique https://github.com/ros-simulation/gazebo_ros_pkgs/pull/783/commits/2bc125b356bcaa39aef6c139c48e697f1f701c5e